File: src/main/java/ru/sfedu/teamselection/repository/TeamRepository.java
package ru.sfedu.teamselection.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import ru.sfedu.teamselection.domain.Team;

import java.util.Optional;

@Repository
public interface TeamRepository extends JpaRepository<Team, Long> {

    boolean existsByNameIgnoreCaseAndCurrentTrackId(String name, Long trackId);

    Optional<Team> findByNameIgnoreCaseAndCurrentTrackId(String name, Long trackId);
}

File: src/main/kotlin/ru/sfedu/teamselection/repository/TeamRepository.kt
package ru.sfedu.teamselection.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.transaction.annotation.Transactional
import ru.sfedu.teamselection.domain.Team

@Transactional
interface TeamRepository: JpaRepository<Team, Long> {
    fun findAllByFullFlagAndCurrentTrackId(isFull: Boolean, trackId: Long): MutableList<Team>
    fun findTeamByTagsInAndCurrentTrackId(tags: List<String>, trackId: Long): MutableList<Team>
    fun findAllByCurrentTrackId(trackId: Long): MutableList<Team>

}